src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -75,23 +75,20 @@ public class ImageTagHelper : UrlResolutionTagHelper
         /// <inheritdoc />
         public override void Process(TagHelperContext context, TagHelperOutput output)
         {
+            output.CopyHtmlAttribute(SrcAttributeName, context);
+            ProcessUrlAttribute(SrcAttributeName, output);
+
             if (AppendVersion)
             {
                 EnsureFileVersionProvider();
 
-                string resolvedUrl;
-                if (TryResolveUrl(Src, encodeWebRoot: false, resolvedUrl: out resolvedUrl))
-                {
-                    Src = resolvedUrl;
-                }
+                // Retrieve the TagHelperOutput variation of the "src" attribute in case other TagHelpers in the
+                // pipeline have touched the value. If the value is already encoded this ScriptTagHelper may
+                // not function properly.
```

Comment:
The only way the attribute could ever explode is if a previous `TagHelper` author did the oddity of: `output.Attributes["src"] = null;`

Essentially I'm being overprotective here and `null` checking below with `?.`. Going to remove the `?.` to not portray that functionality is proper practice.